lib/Backend/SccLiveness.cpp

```diff
@@ -260,9 +260,19 @@ SCCLiveness::Build()
             {
                 this->EndOpHelper(labelInstr);
             }
-            if (labelInstr->isOpHelper && !PHASE_OFF(Js::OpHelperRegOptPhase, this->func))
+            if (labelInstr->isOpHelper)
             {
-                this->lastOpHelperLabel = labelInstr;
+                if (!PHASE_OFF(Js::OpHelperRegOptPhase, this->func) &&
+                    !this->func->GetTopFunc()->GetJITFunctionBody()->IsCoroutine())
+                {
+                    this->lastOpHelperLabel = labelInstr;
+                }
+#ifdef DBG
+                else
+                {
+                    labelInstr->AsLabelInstr()->m_noHelperAssert = true;
```

Comment:
There's an assertion to check that helper blocks are being set correctly - with this optimisation disabled it would fire incorrectly, setting this debug only property disables it.